mica-search/src/main/java/org/obiba/mica/search/queries/AbstractDocumentQuery.java

```diff
@@ -370,23 +375,24 @@ private Properties getJoinFieldsAsProperties() {
       .setSize(scope == DETAIL ? size : 0) //
       .addAggregation(AggregationBuilders.global(AGG_TOTAL_COUNT)); // ;
 
-    if(ignoreFields()) requestBuilder.setNoFields();
+    List<String> sourceFields = getSourceFields();
+
+    if (AGGREGATION == scope) {
+      requestBuilder.setNoFields();
+    } else if (sourceFields != null) {
+      if (sourceFields.isEmpty()) requestBuilder.setNoFields();
+      else requestBuilder.setFetchSource(sourceFields.toArray(new String[0]), null);
```

Comment:
sourceFields.toArray(new String[sourceFields.size()])

Reply:
The String[0] is just to make sure the return is not Object[]

Reply:
In both cases it returns a String[]